scripts/changes-since-last-release.ts

```diff
@@ -0,0 +1,154 @@
+      choices: Object.values(Format),
+      default: Format.PerCommit,
+    })
+    .string('_')
+    .help()
+    .strict()
+    .parse();
+
+  const workspaces = await getWorkspaces();
+  const workspace = workspaces.find((w) => w.name === packageName);
+  if (!workspace) {
+    throw new KnownError(`Could not map ${packageName} to a workspace`);
+  }
+
+  const tags = await getTags();
+  const tagsForWorkspace = tags.filter((line) =>
+    line.startsWith(`${workspace.name}@`),
+  );
+  if (tagsForWorkspace.length === 0) {
+    throw new KnownError(`No version tags found for ${workspace.name}.`);
+  }
+  const versions = tagsForWorkspace.map((tag) => {
+    const versionString = tag.split('@', 3)[2];
+    const parsedVersionString = semver.parse(versionString);
+    if (!parsedVersionString) {
+      throw new Error(`Invalid version tag ${tag}`);
+    }
+    return parsedVersionString;
+  });
+  const sortedVersions = versions.sort((a, b) => a.compare(b));
+  const latestVersion = sortedVersions[sortedVersions.length - 1];
+
+  console.log(
+    `Showing changes for \u001B[34m${
+      workspace.name
+    }\u001B[0m since \u001B[34m${latestVersion.toString()}\u001B[0m.\n`,
+  );
+  await showDiff(workspace, latestVersion, { format });
+}
+
+/**
+ * Retrieve the set of Yarn workspaces in this project.
+ *
+ * @returns The set of workspaces.
+ */
+async function getWorkspaces(): Promise<Workspace[]> {
+  const { stdout } = await execa('yarn', ['workspaces', 'list', '--json']);
+  const workspaces = stdout.split('\n').map((line) => JSON.parse(line));
+  return workspaces.filter((workspace) => workspace.location !== '.');
+}
+
+/**
+ * Retrieve the set of Git tags in this project.
+ *
+ * @returns The set of tags.
+ */
+async function getTags(): Promise<string[]> {
+  const { stdout } = await execa('git', ['tag']);
+  return stdout.split('\n');
+}
+
+/**
+ * Show the set of changes in the latest release of the given package.
+ *
+ * @param workspace - The workspace that represents the package.
+ * @param latestVersion - The version (as a `semver` object).
+ * @param options - An options bag.
+ * @param options.format - How to show the changes: divided by commit or all
+ * together.
+ */
+async function showDiff(
+  workspace: Workspace,
+  latestVersion: semver.SemVer,
+  { format = Format.PerCommit } = {},
+): Promise<void> {
+  const gitOptions = ['--no-pager'];
+  const gitCommandOptions = ['--color'];
+  const args =
+    format === Format.PerCommit
+      ? [
+          ...gitOptions,
+          'log',
+          ...gitCommandOptions,
+          '-p',
+          `${workspace.name}@${latestVersion.toString()}..HEAD`,
+          '--',
+          workspace.location,
+        ]
+      : [
+          ...gitOptions,
+          'diff',
+          ...gitCommandOptions,
+          `${workspace.name}@${latestVersion.toString()}..HEAD`,
+          workspace.location,
+        ];
+  await execa('git', args, {
```

Comment:
This produces a `SIGPIPE (Broken pipe or socket)` error when running this through `less`. I'll work on fixing this.